tasks/utilities/task_validate.wdl

```diff
@@ -54,256 +54,56 @@ task export_two_tsvs {
-    # pd.DataFrame() converts the output of the .apply() function into a Data Frame
-    # .apply(lambda x: function) applys a specific function on each column (x)
-    # validate(x, df1, df2) performs the validation check function
-    # result_type="expand" turns the tuple returned value into a list
-    # .transpose() converts the created DataFrame into a format so it can be added to the summary_output table
-    summary_output[["Validation Criteria", "Number of samples failing the validation criteria"]] = pd.DataFrame(validation_criteria.apply(lambda x: validate(x, df1, df2), result_type="expand")).transpose()
-
-  out_xlsx_name = "~{output_prefix}.xlsx"
-  out_html_name = "~{output_prefix}.html"
-  out_pdf_name = "~{output_prefix}.pdf"
-
-  pd.set_option('display.max_colwidth', None)
-  df_comparison.to_excel(out_xlsx_name)
-
-  # make pretty html table
-  html_table_light_grey = build_table(summary_output, 
-    'grey_light', 
-    index=True, 
-    text_align='center',
-    # conditions={
-    #   'Number of differences (exact match)': {
-    #     'min': 1,
-    #     'max': 0,
-    #     'min_color': 'black',
-    #     'max_color': 'red'
-    #   }
-    # }
-  )
-
-  # save to html file
-  with open(out_html_name, 'w') as outfile:
-    outfile.write("<p>Validation analysis performed on " + str(date.today().isoformat()) + ".</p>")
-    outfile.write(html_table_light_grey)
-    outfile.write("<p>Validation Criteria:</p>")
-    outfile.write("<ul>")
-    outfile.write(" <dt>EXACT</dt>")
-    outfile.write(" <dd>Performs an exact string match</dd>")
-    outfile.write(" <dt>IGNORE</dt>")
-    outfile.write(" <dd>Ignores the values; indicates 0 failures</dd>")
-    outfile.write(" <dt>SET</dt>")
-    outfile.write(" <dd>Compares items in a list without regard to order</dd>")
-    outfile.write(" <dt>PERCENT_DIFF</dt>")
-    outfile.write(" <dd>Tests if two values are more than the indicated percent difference (must be in decimal format)</dd>")
-    outfile.write("</ul>")
-    
-  # convert to pdf
-  options = {
-    'page-size': 'Letter',
-    'title': '~{datatable1} vs. ~{datatable2}',
-    'margin-top': '0.25in',
-    'margin-right': '0.25in',
-    'margin-bottom': '0.25in',
-    'margin-left': '0.25in'
-  }
-
-  output_pdf = pdf.from_file(out_html_name, out_pdf_name, options=options)
-
-  CODE
+    # grab theiavalidate version
+    theiavalidate.py -v > VERSION
+
+    # run theiavalidate
+    theiavalidate.py \
+      ~{datatable1_tsv} \
+      ~{datatable2_tsv} \
+      --columns_to_compare ~{columns_to_compare} \
+      --output_prefix ~{output_prefix} \
+      ~{"--na_values " + na_values} \
+      ~{"--validation_criteria " + validation_criteria_tsv} \
+      ~{"--column_translation " + column_translation_tsv} \
+      ~{true="--debug" false="--verbose" debug_output}
   >>>
   runtime {
-    docker: "us-docker.pkg.dev/general-theiagen/theiagen/utility:1.2"
+    docker: "us-docker.pkg.dev/general-theiagen/theiagen/theiavalidate:0.0.1"
```

Comment:
Any chance we can get a [theiavdalite ](https://github.com/theiagen/theiavalidate/tree/main) v1 release (and associated docker) to use as default before merging this? And, since it's a python package, I'd suggest @cimendes as a reviewer before releasing things.

Reply:
I made a theiavalidate 0.0.1 release and associated docker 👍